kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/UsageSessionServiceAdaptor.java

```diff
@@ -496,9 +496,17 @@ protected void logoutEvent(UsageSession session)
 		
 	}
 
+	// Instance variables to store original user's session information
+	private String originalUserId;
+	private String originalUserEid;
```

Comment:
looks like these variables are class members on a singleton (service), which would mean they're not thread safe.